src/decision/location_selection.jl

```diff
@@ -0,0 +1,151 @@
+
+Perform site selection using a chosen mcda aggregation method, domain, initial cover, criteria weightings and thresholds.
+
+# Arguments
+- `scenario` : contains criteria weightings and thresholds for a single scenario.
+- `mcda_vars` : site selection criteria and weightings structure
+- `w_scens` : array of length nsites containing wave scenario.
+- `dhw_scens` : array of length nsites containing dhw scenario.
+- `site_ids` : locations to consider
+- `sum_cover` : summed cover (over species) for single scenario being run, for each site.
+- `area_to_seed` : area of coral to be seeded at each time step in km^2
+
+# Returns
+- `ranks` : n_reps * sites * 3 (last dimension indicates: site_id, seeding rank, shading rank)
+    containing ranks for single scenario.
+"""
+function site_selection(domain::Domain, scenario::DataFrameRow, w_scens::NamedDimsArray, dhw_scens::NamedDimsArray,
+    site_ids::Vector{Int64}, sum_cover::NamedDimsArray, area_to_seed::Float64)::Matrix{Int64}
+
+    mcda_vars = DMCDA_vars(domain, scenario, site_ids, sum_cover, area_to_seed, w_scens, dhw_scens)
+    n_sites = length(site_ids)
+
+    # site_id, seeding rank, shading rank
+    rankingsin = [mcda_vars.site_ids zeros(Int64, n_sites) zeros(Int64, n_sites)]
+
+    prefseedsites::Vector{Int64} = zeros(Int64, mcda_vars.n_site_int)
+    prefshadesites::Vector{Int64} = zeros(Int64, mcda_vars.n_site_int)
+
+    # Determine connectivity strength
+    # Account for cases where no coral cover
+    in_conn, out_conn, strong_pred = connectivity_strength(domain.TP_data .* site_k_area(domain), Array(sum_cover))
+    in_conn = in_conn[site_ids]
+    out_conn = out_conn[site_ids]
+    strong_pred = strong_pred[site_ids]
+
+    (_, _, ranks) = guided_site_selection(mcda_vars, scenario.guided, true, true, prefseedsites, prefshadesites, rankingsin, in_conn, out_conn, strong_pred)
+
+    return ranks
+end
+
+"""
+    run_site_selection(domain::Domain, scenarios::DataFrame, sum_cover::AbstractArray, area_to_seed::Float64, time_step::Int64)
+
+Perform site selection for a given domain for multiple scenarios defined in a dataframe.
+
+# Arguments
+- `domain` : ADRIA Domain type, indicating geographical domain to perform site selection over.
+- `scenarios` : DataFrame of criteria weightings and thresholds for each scenario.
+- `sum_cover` : array of size (number of scenarios * number of sites) containing the summed coral cover for each site selection scenario.
+- `area_to_seed` : area of coral to be seeded at each time step in km^2
+- `target_seed_sites` : list of candidate locations for seeding (indices)
+- `target_shade_sites` : list of candidate location to shade (indices)
+
+# Returns
+- `ranks_store` : number of scenarios * sites * 3 (last dimension indicates: site_id, seed rank, shade rank)
+    containing ranks for each scenario run.
+"""
+function run_site_selection(dom::Domain, scenarios::DataFrame, sum_cover::AbstractArray, area_to_seed::Float64; aggregate_function=nothing,
```

Comment:
The arguments don't match the Docstring (both at the name of the function and at the list of arguments)